Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -247,6 +248,34 @@ public struct TargetSourcesBuilder {
         }
     }
 
+    private func diagnoseMissingResources(in resources: [Resource]) {
+        // Stop early if the valid resources array count is the same as the user defined list
+        if resources.count == self.target.resources.count {
+            return
+        }
+        
+        // Filter the package list of resources down to those which are not included in the valid resources array
+        let invalidResources = self.target.resources.filter { targetResource in
+            return resources.contains(where: { resource in
+                resource.path.relative(to: self.targetPath).pathString.hasPrefix(targetResource.path)
```

Comment:
By converting to a string here, it would match directories whose name happens to be a prefix of another name.  But I think you can use `AbsolutePath.contains()` to just compare path components.

Reply:
This logic was broadly copied from other logic within the file but I'm more than happy to explore other ways of carrying out the same thing 👍 

I'll investigate this.

Reply:
Thank you!  I see — yes, in that case it looks as if the other logic should be corrected as well (not as part of this PR, of course, and certainly not necessarily by you).  I'll take a look at what I would suggest as the right expression here involving `AbsolutePath.contains()`.  Probably it would be to construct the absolute path for the `targetResources` and then to check if that `.contains()` the path of the `resource` itself.

Reply:
I've taken a look at this, and I think that this:
`         let invalidResources = self.target.resources.filter { targetResource in
             let targetResourcePath = self.targetPath.appending(RelativePath(targetResource.path))
             return resources.contains(where: { resource in
                targetResourcePath.contains(resource.path)
             }) == false
          }
`
would work instead.  The key here is that the `.contains()` is a method on AbsolutePath and not on String.  It might be worth considering better names for the local variables here, though, to make it clear what's being compared.

Reply:
I'm currently on a two week vacation but I'll be sure to take a look at this once I return -- I can definitely see where you're going with the sample above so let me try it out and see where that takes us.

Thanks for the help!

Reply:
Thanks a lot, @sherlouk, and enjoy your vacation!

Reply:
Taken a look into this today and seems with the suggestion above we're (still?) running into problems identifying entire directories.

So if you define a resource as a path pointing to a directory, even if the directory has files in it - the check raises it as invalid (which of course isn't true!)

Do you have any ideas on how we could resolve this?

Thanks 😄

Reply:
Thanks for the additional changes — I'm not that familiar with the underlying code but will make some time to take a look at this today or tomorrow and see what I can find out.  Thanks!